File: target_includes/android_bridges/Application.java
package de.fhflensburg.tobiasmarkus.androidBridge;

import android.graphics.Point;
import android.view.Display;

public class Application {
	
	/**
	 * The android activity
	 */
	private android.app.Activity innerActivity = null;
	
	/**
	 * The current "bridge" activity
	 */
	private ApplicationView applicationView = null;
	
	/**
	 * The height and width of the created activity
	 */
	private int width = 0, height = 0;
	
    /**
     * Initialize window
     * @param context The main Activity (aka main function) this window is wrapped in
     */
	public Application(android.app.Activity context) {
		this.innerActivity = context;
		
		Display display = innerActivity.getWindowManager().getDefaultDisplay();
		
		// Display-Size-Variable anlegen:
		Point display_size = new Point();

		// Groesse des Bildschirms abholen:
		display.getSize(display_size);
		
		// Speichern der Groesse in den Variablen width und height:
		this.width = display_size.x;
		this.height = display_size.y;
	}
	
	/**
	 * Gets the height of this application
	 * @return Height of this application in pixels
	 */
	public int getHeight()
	{
		return height;
	}
	
	/**
	 * Gets the width of this application
	 * @return Width of this application in pixels
	 */
	public int getWidth()
	{
		return width;
	}
	
    /**
     * Return the wrapped Activity instance
     * @return Activity The wrapped Activity instance
     */
	public android.app.Activity getWrappedElement() {
		return innerActivity;
	}

    /**
     * Create an activity inside this window
     * @return The created activity
     */
	public ApplicationView createApplicationView() {
		applicationView = new ApplicationView(innerActivity, width, height);
		return applicationView;
	}
	
	/**
	 * Gets the current activity inside this window
	 * @return The activity inside this window
	 */
	public ApplicationView getApplicationView() {
		return applicationView;
	}
}

File: target_includes/ios_bridges/Application.swift
//
//  Application.swift
//  MobileApplicationsSwiftAufgabe1
//
//  Created by Tobias Markus on 21.04.15.
//  Copyright (c) 2015 Tobias Markus. All rights reserved.
//

import Foundation
import UIKit

/**
 * Class that represents an iOS Application Window
 */
public class Application {
    
    /**
     * Variable that saves the parent UIResponder object (for later reference)
     */
    var parentContext: UIResponder;
    
    /**
     * Wrapped UIWindow instance
     */
    var parentWindow: UIWindow;
    
    /**
     * Initialize window
     * @param context The UI Responder object (aka main function) we create this activity for
     */
    public init(context: UIResponder) {
        parentContext = context;
        
        // Größe festlegen:
        parentWindow = UIWindow(frame: UIScreen.mainScreen().bounds);
        
        
        // Hintergrundfarbe festlegen:
        parentWindow.backgroundColor = UIColor.whiteColor();
    }
    
    /**
     * Initialize window
     * @param context The UI Responder object (aka main function) we create this activity for
     * @param width Width of this activity
     * @param height Height of this activity
     */
    public init(context: UIResponder, width: CGFloat, height: CGFloat)
    {
        parentContext = context;
        
        // Größe festlegen:
        parentWindow = UIWindow(frame: CGRectMake(0, 0, width, height));

        // Hintergrundfarbe festlegen:
        parentWindow.backgroundColor = UIColor.whiteColor();
    }
    
    /**
     * Function call required by iOS
     */
    required public init(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
    /**
     * Return the wrapped UIWindow instance
     * @return UIWindow The wrapped UIWindow instance
     */
    public func getWrappedElement() -> UIWindow {
        return parentWindow;
    }
    
    /**
     * Create an activity inside this window
     * @return The created activity
     */
    public func createApplicationView() -> ApplicationView {
        var activity = Activity(context: parentContext);
        parentWindow.rootViewController = activity;
        return activity;
    }
    
    /**
     * Returns the current activity
     * @return the activity belonging to this window
     */
    public func getApplicationView() -> ApplicationView
    {
        return parentWindow.rootViewController as! ApplicationView;
    }
}